### DISCUSSION TITLE
Implementing API Key Authentication with NextAuth in a Vercel/NextJS App

### DISCUSSION THREAD

**Discussion by KentonParton** [nextauthjs/next-auth#8471] [Help]
Hello Everyone,

We're utilizing NextAuth for authentication on our platform at https://app.kwip.ai/ and are interested in incorporating API Key-based authentication as well. Does anyone have recommendations on how to implement API Key authentication in conjunction with NextAuth?

If NextAuth isn't suitable for this, do you have any alternative suggestions specifically for a Vercel/NextJS environment?

Thank you!

**Comment by abhay-ramesh** [comment]
Any progress on this?

**Comment by lancejpollard** [comment]
Yeah I was just looking at https://clerk.com/pricing and they too don't offer API key management, does anyone do this as a service even?
